src/main/java/io/conduit/DestinationConfig.java

```diff
@@ -17,32 +17,103 @@
+    private static final List<String> REQUIRED_KEYS = List.of(
+        "catalog.name", "namespace", "table.name",
+        "s3.endpoint", "s3.access-key-id", "s3.secret-access-key"
+    );
+
+    @JsonProperty(value = "catalog.name")
+    private String catalogName;
+
+    @JsonProperty(value = "namespace")
+    private String namespace;
+
+    @JsonProperty(value = "table.name")
+    private String tableName;
+
+    @JsonProperty(value = "s3.endpoint")
+    private String s3Endpoint;
+
+    @JsonProperty(value = "s3.access-key-id")
+    private String s3AccessKeyId;
+
+    @JsonProperty(value = "s3.secret-access-key")
+    private String s3SecretAccessKey;
+
+    private Map<String, String> catalogProperties = new HashMap<>();
+
     /**
      * Creates a new <code>DestinationConfig</code> instance from a map with configuration parameters.
+     * The map is required to have:
+     * <li><code>namespace</code></li>
+     * <li><code>table.name</code></li>
+     * <li><code>catalog.name</code></li>
+     * <p>
+     * The catalog properties need to prefixed with <code>catalog.catalog_name</code>.
+     * If, for example, the catalog's name is <code>ProdCatalog</code>, and it has a
+     * parameter <code>uri=https://example.com</code>, then the following should be added to the map:
+     * <code>catalog.ProdCatalog.uri=https://example.com</code>
+     *
+     * @throws IllegalArgumentException if required keys are missing or if there are unknown parameters.
      */
-    public static DestinationConfig fromMap(Map<String, String> map) {
-        if (Utils.isEmpty(map)) {
-            return new DestinationConfig();
+    public static DestinationConfig fromMap(Map<String, String> cfgMap) {
+        checkRequired(cfgMap);
+
+        DestinationConfig cfg = mapper.convertValue(cfgMap, DestinationConfig.class);
+
+        var unknownProps = cfg.getCatalogProperties().keySet().stream()
+            .filter(not(k -> k.startsWith(CATALOG_PREFIX + cfg.getCatalogName() + ".")))
+            .toList();
+        if (!Utils.isEmpty(unknownProps)) {
+            throw new IllegalArgumentException("unknown properties: " + unknownProps);
         }
-        return fromMap(map, DestinationConfig.class);
+
+        return cfg;
     }
 
-    protected static <T extends DestinationConfig> T fromMap(Map<String, String> map, Class<T> clazz) {
-        Map<String, String> connectorMap = new HashMap<>();
+    private static void checkRequired(Map<String, String> cfgMap) {
+        List<String> missing = REQUIRED_KEYS.stream()
+            .filter(not(cfgMap::containsKey))
+            .toList();
 
-        map.forEach(connectorMap::put);
-        T cfg = Utils.mapper.convertValue(connectorMap, clazz);
+        if (!Utils.isEmpty(missing)) {
+            throw new IllegalArgumentException("missing keys: " + missing);
+        }
+    }
 
-        return cfg;
+    // Capture all other fields that Jackson do not match other members
```

Comment:
do you mean Json? I'm not sure I understand this comment tbh

Reply:
removed comment